internal/controller/apiextensions/claim/configurator.go

```diff
@@ -92,7 +94,22 @@ func (c *APIDryRunCompositeConfigurator) Configure(ctx context.Context, cm resou
 	// external name.
 	en := meta.GetExternalName(ucp)
 
-	meta.AddAnnotations(ucp, ucm.GetAnnotations())
+	// Do not propagate *.kubernetes.io annotations down to the composite
+	// For example: when a claim gets deployed using kubectl,
+	// its kubectl.kubernetes.io/last-applied-configuration annotation
+	// should not be propagated to the corresponding composite resource,
+	// because:
+	// * XR was not created using kubectl
+	// * The content of the annotaton refers to the claim, not XR
+	// See https://kubernetes.io/docs/reference/labels-annotations-taints/
+	// for all annotations and their semantic
+	annot := ucm.GetAnnotations()
+	maps.DeleteFunc(annot, func(k string, _ string) bool {
```

Comment:
https://cs.opensource.google/go/x/exp

> This subrepository holds experimental and deprecated (in the old directory) packages.

This doesn't look ideal. Should we instead implement that simple functionality (delete keys from a map) instead of relying on an experimental package?

Reply:
That library was already project dependency, so I though it is ok to use it. Reverted.

Reply:
Not sure how it ended up being there but couldn't find any previous usage in crossplane or crossplane-runtime.